ompi/mca/mtl/psm2/mtl_psm2_component.c

```diff
@@ -389,6 +378,27 @@ ompi_mtl_psm2_component_init(bool enable_progress_threads,
         ompi_mtl_psm2_set_shadow_env (ompi_mtl_psm2_shadow_variables + i);
     }
 
+#if OPAL_CUDA_SUPPORT
+    /*
+     * If using CUDA enabled OpenMPI, the user likely intends to
+     * run with CUDA buffers. So, force-set the envvar here if user failed
+     * to set it.
+     */
+    ret = glob("/sys/module/nvidia", GLOB_DOOFFS, NULL, &globbuf);
+    if (0 == ret ||  GLOB_NOMATCH == ret) {
```

Comment:
Unless we set `GLOB_NOCHECK` flag, I think we can expect `globbuf.gl_pathv` to be empty if the pattern didn't match. (I verified this on a sandbox too). So, I'll modify this to check:
    if (globbuf.gl_argc > 0) {
        globfree(&globbuf)
    }

FYI- The current check I have was basically following the existing usage in lines 259, 264. Which itself seems to have been introduced in commit 1daa80d (plug a memory leak in ompi_mtl_psm2_component_open). I don't think this commit was cherry-picked to other branches though. If this is fixing a memory leak, should this commit be ported to v2.x, v3.0.x?

While at it, shall I modify the above usage to also initialize the entire struct to `{0}` and to `globfree()` only `if (globbuf.gl_argc > 0)`?

Reply:
I think the answer is "yes" to all your questions:

* Cherry pick as relevant to v2.x, v3.0.x, and v3.x (since this is fixing a [minor] bug).
    * Skip v2.0.x -- that series is effectively dead.  We're really only taking serious bug fixes to that series now (this minor memory leak is not serious enough).
* I like your idea of initializing the entire struct with `{0}` and only `globfree()`ing if `globbuf.gl_argc > 0`.

Reply:
Fixed this (Y)